src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,99 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+import { FC, useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[];
+  setFilteredTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+};
+
+enum SortType {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
+
+const todoFilter = (
+  todos: Todo[],
+  sortType: string,
+  query: string,
+): Todo[] => {
+  let filteredTodos = [...todos];
+
+  switch (sortType) {
+    case SortType.Active:
+      filteredTodos = filteredTodos.filter(todo => !todo.completed);
+      break;
+
+    case SortType.Completed:
+      filteredTodos = filteredTodos.filter(todo => todo.completed);
+      break;
+
+    default:
+      break;
```

Comment:
Do not leave the `default` block empty, there you can throw an error about the wrong type of sorting

Reply:
> Do not leave the `default` block empty, there you can throw an error about the wrong type of sorting

the code doesn't compile with
```
  default:
    throw new Error('Цrong type of sorting!');
```

Reply:
@errisi I can't tell you anything about compilation problems without error text or a screenshot, but the option to leave the default block empty is definitely wrong.